integration/manifests/cleaner/cleaner-for-calico.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: danm-cleaner
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: caas:danm-cleaner
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+  - watch
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - events
+  - endpoints
+  verbs:
+  - create
+  - update
+  - get
+- apiGroups:
+  - danm.k8s.io
+  resources:
+  - danmnets
+  - danmeps
+  - tenantnetworks
+  - clusternetworks
+  verbs: [ "*" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: caas:danm-cleaner
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: caas:danm-cleaner
+subjects:
+- kind: ServiceAccount
+  name: danm-cleaner
+  namespace: kube-system
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: danm-cleaner
+  namespace: kube-system
+  labels:
+    danm: cleaner
+spec:
+  selector:
+    matchLabels:
+     danm: cleaner
+  template:
+    metadata:
+      name: danm-cleaner
+      labels:
+        danm: cleaner
+    spec:
+      serviceAccountName: danm-cleaner
+      containers:
+      - name: danm-cleaner
+        image: cleaner
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: calico-config
+          mountPath: /etc/calico/calicoctl.cfg
+          readOnly: true
+        - name: calicoctl
```

Comment:
package dependencies into containers
that's what they are for :)

Reply:
In that case what should happen if bundled version of calicoctl is API incompatible with calico control plane?
I am mounting calicoctl from host because is should be API compatible with Calico control plane and the binary will not consume space in the container image. Do you still want it bundled instead?

Reply:
never ever should you expect to mount anything from host
that's an anti-pattern

Reply:
implement programmatic communication with the Calico API if oyu are worried about compatibility. the CLI also uses... something I guess

Reply:
I tried the programmatic way, but got a bunch of API incompatibilities. Eventually I went with downloading calicoctl binary into cleaner image. I removed host mounting calicoctl from manifest.

Reply:
looks like the CLI actually talks to ETCD directly?
add a TODO to follow up on this. maybe we only need to talk to ETCD REST API directly in this case, and skip Calico control plane
but ye, okay like this rn